calico-enterprise/network-policy/domain-based-policy.mdx

```diff
@@ -229,6 +229,13 @@ spec:
       destination:
         selector: color == 'red'
 ```
+## Staged DNS Policy
+
+If you define a staged DNS policy the rule to allow egress on the domains specifid in the staged policy or the networkset
+used is not always matched. This is because of the hostname to ip address conversion for the first time you try to reach
+to the domain address and the ttl for that conversion. This will not happen for enforced policies, but if you prefer to
```

Comment:
I think this can be expressed better, and with a narrower scope - making clear that it only applies to connection setups just after a DNS lookup - as follows:

"When a pod first connects to a domain that should be allowed by DNS policy, you may observe that the first connection setup packet is denied.  This is because {{site.prodname}} is still updating iptables or eBPF programming to allow the IP addresses that were indicated in the DNS lookup response, _at the same time_ as the pod is connecting to one of those IP addresses.  If the connection is TCP-based - such as an HTTP request - the TCP level will automatically send another connection setup packet after 1 second, and that packet should be allowed (unless the node concerned is so heavily loaded that {{site.prodname}} is struggling to get any CPU!), which means that the connection is established after a latency of 1 second.  If the connection is UDP-based, it is up to the application to retry connection setup - i.e. what happens will depend on the detailed coding of the particular application that the pod is using.

"You can eliminate the above concerns by setting `DNSPolicyMode` in `FelixConfiguration` to `DelayDNSResponse`.  This works by ensuring that the pod doesn't see the DNS lookup response - and so doesn't try to connect to one of its IP addresses - until {{site.prodname}} has fully updated iptables or eBPF to allow those IP addresses.  But it has a small downside of adding a slight latency to every DNS lookup that pods make, and that is why it is not the default setting of `DNSPolicyMode`."

Reply:
Thanks @nelljerram - One more thing that I wanted to clarify here was that why the behaviour between staged and enforces dns policies are different. Do you think it would help if we first give them a high level explanation and then provide a more elaborate technical explanation?